### DISCUSSION TITLE
how to build source code?

### DISCUSSION THREAD

**Discussion by shangjianan2** [rabbitmq/rabbitmq-java-client#1392] [General]
### Describe the bug

I want to build a new version, but after downloading the source code, I'm unable to compile it. (Currently, the known problem is that there is no class named "com.rabbitmq.client.AMQP" in the source code, but this class exists in "com.rabbitmq:amqp-client"). Could you please tell me how to solve this problem?

![image](https://github.com/user-attachments/assets/c24608c0-0599-4ae5-89f1-a971a8c4340f)

![image](https://github.com/user-attachments/assets/0bbdd9f1-d8e7-4f38-bd96-b01c4917f642)
![image](https://github.com/user-attachments/assets/8b4b729a-77e4-436a-be0f-da6b3bafc565)
![image](https://github.com/user-attachments/assets/a7c0b63c-59b2-4dad-82ee-4d0abe0bd263)

### Reproduction steps

1.git clone https://github.com/rabbitmq/rabbitmq-java-client.git
2.Use idea new project from existing sources
3.compile
4.error: "Failed to generate D:\data\BianCheng\java\rabbitmq\source\rabbitmq-java-client\target/generated-sources/src/main/java/com/rabbitmq/client/AMQP.java with command: python D:\data\BianCheng\java\rabbitmq\source\rabbitmq-java-client/codegen.py header D:\data\BianCheng\java\rabbitmq\source\rabbitmq-java-client/deps/rabbitmq_codegen/amqp-rabbitmq-0.9.1.json D:\data\BianCheng\java\rabbitmq\source\rabbitmq-java-client\target/generated-sources/src/main/java/com/rabbitmq/client/AMQP.java
"
...

### Expected behavior

compile success

### Additional context

jdk1.8

**Comment by michaelklishin** [comment]
This client uses code generation, see https://www.rabbitmq.com/client-libraries/build-java-client
